### PR TITLE
Add accessors to LiteralKind and make value/floatval private

### PR DESCRIPTION
**Pull request by jvilk-stripe**
Add accessors to LiteralKind. Makes it safer to extract the inner double/int/NameRef contained within.

### Motivation

Delurk lurky code that directly accesses LiteralType's value and floatval properties without the benefit of ENFORCEs checking that the literal is the expected kind of literal.

### Test plan

Covered by existing tests.

### CODE REVIEW

File: core/types/types.cc [hunk 1]
```diff
@@ -334,6 +334,21 @@ LiteralType::LiteralType(SymbolRef klass, NameRef val)
...
+}
+
+double LiteralType::asFloat() const {
+    ENFORCE_NO_TIMER(literalKind == LiteralTypeKind::Float);
+    return floatval;
+}
+
+core::NameRef LiteralType::asName() const {
+    ENFORCE_NO_TIMER(literalKind == LiteralTypeKind::Symbol || literalKind == LiteralTypeKind::String);
+    return NameRef(NameRef::WellKnown{}, value);
```
Comment (jez) [inline core/types/types.cc:349]: This seems wrong, doesn't it? shouldn't we pass a GlobalState from the call site?

Comment (jvilk-stripe) [reply core/types/types.cc:349]: Sure, I can do that.

Comment (jez) [reply core/types/types.cc:349]: (every call site this replaced seems to have a GlobalState in scope directly)

File: core/types/types.cc [hunk 2]
```diff
@@ -367,15 +382,17 @@ void LiteralType::_sanityCheck(const GlobalState &gs) const {
 }
 
 bool LiteralType::equals(const LiteralType &rhs) const {
-    if (this->value != rhs.value) {
```
Comment (jez) [inline core/types/types.cc:370]: oh no was this doing bitwise floating point comparison

**Review by jez — APPROVED** [review]
Looks good to me except the GlobalState comment.

File: core/types/types.cc (hunk 2 above)
Comment (jvilk-stripe) [reply core/types/types.cc:370]: SURE WAS

File: core/types/types.cc (hunk 1 above)
Comment (trevor-stripe) [reply core/types/types.cc:354]: The only place that would be complicated by that is the pickling code, as it used to just always encode the `u8` value of the union.

Comment (jvilk-stripe) [reply core/types/types.cc:354]: Fixed. I did have to add an `unsafe` variant for the serializer callsite.

Comment (jez) [reply core/types/types.cc:354]: Gotcha, so at that point, we could pass WellKnown, and write a comment about why it doesn't matter that we're using WellKnown for that call site.

**Pull request merged**